sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java

```diff
@@ -655,6 +655,22 @@ AccessToken getTokenFromAzureCLIAuthentication(StringBuilder azCommand) {
         return token;
     }
 
+    static OffsetDateTime getTokenExpiryOffsetDateTime(Map<String, String> objectMap) {
```

Comment:
nit: maybe this is two helper methods: one to parse `expires_on`, one to parse `expiresOn`. That way you can test and detect that we are properly detecting when `expires_on` isn't present. Right now there's no way to assert which path through this method we took.